extensions/wikia/ThemeDesigner/ThemeSettings.class.php

```diff
@@ -124,12 +130,59 @@ public function getHistory() {
 		}
 
 		foreach ( $history as $key => $val ) {
-			$history[$key]['settings']['background-image'] = $this->getFreshURL( $val['settings']['background-image-name'], ThemeSettings::BackgroundImageName );
+			$history[$key]['settings']['background-image'] = $this->getFreshURL(
+				$val['settings']['background-image-name'],
+				ThemeSettings::BackgroundImageName
+			);
+			$history[$key]['settings']['community-header-background-image'] = $this->getFreshURL(
+				$val['settings']['community-header-background-image'],
+				ThemeSettings::CommunityHeaderBackgroundImageName
+			);
 		}
 
 		return $history;
 	}
 
+	private function saveImage(string $name, string $title, bool $setSize = false, bool $setUserImage = false) {
+		if ( isset( $settings["${$name}-name"] ) && strpos( $settings["${$name}-name"], 'Temp_file_' ) === 0 ) {
+			$temp_file = new LocalFile( Title::newFromText( $settings["${$name}-name"], NS_FILE ), RepoGroup::singleton()->getLocalRepo() );
+			$file = new LocalFile( Title::newFromText( $title, NS_FILE ), RepoGroup::singleton()->getLocalRepo() );
+			$file->upload( $temp_file->getPath(), '', '' );
+			$temp_file->delete( '' );
+
+			$settings["${$name}"] = $file->getURL();
+			$settings["${$name}-name"] = $file->getName();
+
+			if ($setSize) {
+				$settings["${$name}-width"] = $file->getWidth();
+				$settings["${$name}-height"] = $file->getHeight();
+			}
+
+			if ($setUserImage) {
+				$imageServing = new ImageServing( null, 120, [ "w" => "120", "h" => "65" ] );
+				$settings["user-${$name}"] = $file->getURL();
+				$settings["user-${$name}-thumb"] = wfReplaceImageServer(
+					$file->getThumbUrl(
+						$imageServing->getCut(
+							$file->getWidth(),
+							$file->getHeight(),
+							"origin"
+						) . "-" . $file->getName()
+					)
+				);
+			}
+
+			$file->repo->forceMaster();
+			$history = $file->getHistory( 1 );
+
+			if ( count( $history ) == 1 ) {
+				return [ 'url' => $history[0]->getURL(), 'name' => $history[0]->getArchiveName() ];
+			}
+		}
+
+		return null;
+	}
+
 	public function saveSettings( $settings, $cityId = null ) {
 		global $wgCityId, $wgUser;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)